### PR TITLE
Add source-canvas documentation

### PR DESCRIPTION
**Pull request by lbud**
Closes https://github.com/mapbox/mapbox-gl-style-spec/issues/659.

image

### CODE REVIEW

File: docs/_generate/index.html [hunk 1]
```diff
@@ -488,6 +489,67 @@ <h3 class='space-bottom1'><a href='#sources-video' title='link to video'>video</
             </tbody>
           </table>
         </div>
+        <div id='sources-canvas' class='pad2 keyline-bottom'>
+          <h3 class='space-bottom1'><a href='#sources-canvas' title='link to canvas'>canvas</a></h3>
+          <p>
+            A canvas source. The <code>"canvas"</code> value is the ID of the canvas element in the document.
```
Comment (1ec5) [inline docs/_generate/index.html:495]: This is already in a `<code>` tag, so it doesn't need quotation marks. (×2)

File: docs/_generate/index.html [hunk 2]
```diff
@@ -488,6 +489,67 @@ <h3 class='space-bottom1'><a href='#sources-video' title='link to video'>video</
...
+              <td class='center'>Android SDK</td>
+              <td class='center'>iOS SDK</td>
+              <td class='center'>macOS SDK</td>
+            </tr>
+            </thead>
+            <tbody>
+            <tr>
+              <td>basic functionality</td>
+              <td>&gt;= 0.32.0</td>
+              <td><a href="https://github.com/mapbox/mapbox-gl-native/issues/7823">Not yet supported</a></td>
```
Comment (1ec5) [inline docs/_generate/index.html]: mapbox/mapbox-gl-native#7823 only applies to the iOS and macOS SDKs; I have no idea whether it makes sense on Android. Moreover, it's unlikely that any gl-native feature will look anything like this canvas source type. "Not supported" would be more accurate.

Comment (1ec5) [reply docs/_generate/index.html]: "Not supported" should be unlinked as well.

**Review by 1ec5 — CHANGES_REQUESTED** [review]

File: docs/_generate/index.html (hunk 1 above)
Comment (lbud) [reply docs/_generate/index.html:495]: Hmm…much of this documentation _does_ contain `<code>"something"</code>` — it seems to generally apply to many JSON values https://www.mapbox.com/mapbox-gl-style-spec/#sources. I copied this style from video sources, right above it ([e.g.](https://github.com/mapbox/mapbox-gl-style-spec/blob/mb-pages/docs/_generate/index.html#L439-L444))

Comment (1ec5) [reply docs/_generate/index.html:495]: Ah, I guess this is an inconsistency between the source and layer documentation. Fodder for other PRs, I guess.

**Review by 1ec5 — APPROVED** [review]

**Pull request merged**